tool/update_github_emojis.dart

```diff
@@ -0,0 +1,338 @@
+  }
+  if (visualizeUnicodeDiffs) {
+    print(
+        'The following emojis have different unicode sequences from those of legacy versions:');
+  }
+  final client = HttpClient();
+  final request = await client.getUrl(Uri.parse(_emojisJsonRawUrl));
+  final response = await request.close();
+  final Map<String, String> shortcodeToEmoji = jsonDecode(
+          await response.cast<List<int>>().transform(utf8.decoder).join(''))
+      .map((String alias, dynamic filename) => MapEntry(
+          alias, parseGithubFilenameIntoUnicodeString(filename as String)))
+      .cast<String, String>() as Map<String, String>;
+
+  // Now before we proceed we need to 'mix in' any legacy emoji alias shortcodes that
+  // are missing from the github emoji list.
+  legacyEmojis.forEach((String shortCodeAlias, String emojiUnicode) {
+    if (!shortcodeToEmoji.containsKey(shortCodeAlias)) {
+      shortcodeToEmoji[shortCodeAlias] = emojiUnicode;
+    }
+  });
+
+  final emojisContent = StringBuffer('''
+// GENERATED FILE. DO NOT EDIT.
+//
+// This file was generated from Githubs api emoji list end point:
+// $_emojisJsonRawUrl
+// at ${DateTime.now()} by the script, tool/update_github_emojis.dart.
+
+''');
+  emojisContent.writeln('const emojis = <String, String>{');
+  var emojiCount = 0;
+  final ignored = <String>[];
+  final errored = <String>[];
+  // Dump in sorted order now to facilitate comparison with new github emojis.
+  final sortedKeys = shortcodeToEmoji.keys.toList()..sort();
+  for (final String shortCodeAlias in sortedKeys) {
+    String emojiUnicode = shortcodeToEmoji[shortCodeAlias]!;
+    if (useLegacyUnicodeSequences &&
+        legacyEmojis.containsKey(shortCodeAlias) &&
+        shortCodeAlias != 'cricket' &&
+        shortCodeAlias != 'beetle') {
+      emojiUnicode = legacyEmojis[
+          shortCodeAlias]!; // Use legacy unicode string if available
+    }
+    if (legacyEmojis.containsKey(shortCodeAlias) &&
+        emojiUnicode != legacyEmojis[shortCodeAlias]) {
+      totalEmojisWithDifferentUnicodeSequences++;
+      if (visualizeUnicodeDiffs) {
+        print(
+            '$emojiUnicode was ${legacyEmojis[shortCodeAlias]} :$shortCodeAlias:');
+      }
+    }
+    if (emojiUnicode != errorSpecialReplacement && emojiUnicode.isNotEmpty) {
+      emojisContent.writeln("  '$shortCodeAlias': '$emojiUnicode',");
+      if (dumpMarkdownShortCodes) {
+        print(':$shortCodeAlias:');
+      } else if (dumpMarkdownToolTipShortCodes) {
+        print('[:$shortCodeAlias:](## "&colon;$shortCodeAlias&colon; emoji")');
+      }
+      emojiCount++;
+    } else {
+      if (emojiUnicode == errorSpecialReplacement) {
+        errored.add(shortCodeAlias);
+      } else {
+        ignored.add(shortCodeAlias);
+      }
+    }
+  }
+  emojisContent.writeln('};');
+  File(_emojisFilePath).writeAsStringSync(emojisContent.toString());
+
+  if (dumpMarkdownShortCodes) {
+    // No report so we don't put junk in our markdown we have
```

Comment:
Hmm I don't super understand this comment. Where could we be putting junk? Just stdout?

Reply:
That flag dumps some markdown that can be used to test all emojis.  Here we exit early before summary is written so that we don't clutter the markdown that was going to stdout (and presumably being captured)

I have changed the comment to 
```
    // We are outputing the markdown to stdout, and presumably it
    // is being captured, so we exit now to exclude the summary
    // report from being included in the emoji markdown we have 
    // been outputing.
```